apps/fileworker/src/lib/db/schema.ts

```diff
@@ -0,0 +1,32 @@
+import { sql } from 'drizzle-orm'
+import { int, sqliteTable, text } from 'drizzle-orm/sqlite-core'
+import { createInsertSchema, createSelectSchema } from 'drizzle-zod'
+import { customAlphabet } from 'nanoid'
+
+import type { z } from 'zod'
+
+export const files = sqliteTable('files', {
+	file_id: int().primaryKey({ autoIncrement: true }),
+	file_external_id: text()
+		.notNull()
+		.unique() // in case of very rare collisions
+		.$defaultFn(() => createId()),
+	name: text().notNull(),
+	created_on: text()
+		.notNull()
+		.default(sql`(CURRENT_TIMESTAMP)`),
+	expires_on: text().notNull(),
+})
+
+// Zod schemas for the table. Docs: https://orm.drizzle.team/docs/zod
+
+export type FileSelect = z.infer<typeof FileSelect>
+export const FileSelect = createSelectSchema(files)
+
+export type FileInsert = z.infer<typeof FileInsert>
+export const FileInsert = createInsertSchema(files)
+
+const nanoid = customAlphabet('0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz')
+function createId(): string {
+	return nanoid(9)
```

Comment:
creates 9 character random IDs - should be plenty random to rely on the unique constraint in the DB to prevent conflicts. If it somehow did conflict, the upload will fail anyway so no need to check if an ID is in use first.